### DISCUSSION TITLE
build keeps failing due to SK imports

### DISCUSSION THREAD

**Discussion by ioucyf** [sveltejs/kit#9362] [Q&A]
I never had a problem with the `npm run build` command.
I also do not have any problems with the `npm run dev` command.
Everything works as expected.

Now however,
`npm run build` keeps throwing these errors, and I cannot figure out why...
yet, `npm run dev` works fine still.

```
[vite]: Rollup failed to resolve import "$app/environment" from "/node_modules/@personal-module/lib/mod.js"
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`


[vite]: Rollup failed to resolve import "$env/static/public" from "/src/lib/mod.ts".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
```

Any ideas?

**Comment by kyllerss** [comment]
@ofeenee , are you still running into this issue? Did you find the solution?

If not, you should provide an example and more information so the maintainers can triage it and provide some guidance.

**Reply by ioucyf** [reply]
Hello @kyllerss,

Honestly, I do not know if the problem still exists or not. I'll test it tomorrow.

However, to describe the problem: I made an npm private package library for some things (components, stores, etc) that I always use for my different projects. This way, I get to update the files once, update the project via npm from each project to get the latest updates. The problem occurs when any of the package files contains the SvelteKit's builtin aliases such as `"$env/static/public"` and `"$app/environment"`. 

Right now, I copy/paste those same files into each project's lib directory, and whenever I update the code of these files, I copy/paste the files from and to all the other projects, manually. It sucks, but when I do it this way, everything works fine.

I have to revert the change, reuse my private package, and test if the problem still exists or not.


The weird thing about this issue is that in development, everything runs perfectly well. It only occurs when I do `npm run build`.
